src/app/beer_garden/api/http/base_handler.py

```diff
@@ -250,3 +251,29 @@ def schema_validated_body(self, schema: Type[Schema]) -> dict:
             return schema(strict=True).load(self.request_body).data
         except MarshmallowValidationError:
             raise BadRequest
+
+    def validate_or_raise(
```

Comment:
This is similar, in spirit, to `schema_validated_body` but adds some useful features:

1. It is more general. It works on more than just the request body which was necessary in the new endpoint code
2. It raises a specific HTTP exeception: `ValidationError` defined in `beer_garden.api.http.exceptions`. Moreover, it preserves passed messages for the `reason`, which is expected to be useful to clients. In particular, the UI code, but also the `http_client` used in unit tests.